interop-cats/shared/src/main/scala/zio/interop/CHub.scala

```diff
@@ -149,51 +150,51 @@ object CHub {
    *
    * For best performance use capacities that are powers of two.
    */
-  def bounded[F[+_]: Effect, A](requestedCapacity: Int)(implicit runtime: Runtime[Any]): F[Hub[F, A]] =
-    toEffect(ZHub.bounded[A](requestedCapacity).map(hub => CHub(hub)))
+  def bounded[F[+_]: Async: Dispatcher, A](requestedCapacity: Int)(implicit runtime: Runtime[Any]): F[Hub[F, A]] =
+    ZHub.bounded[A](requestedCapacity).map(hub => CHub(hub)).toEffect[F]
```

Comment:
The type parameter in `toEffect` here and below can be elided, right?

Reply:
No, looks like type inference doesn't quite manage

Reply:
Really? Which version?

Reply:
Scala 2.13 - I wonder if it matters that `F[+_]` is covariant - we don't actually need it here.
Maybe this will go away if we take `Async` and `Dispatcher` as constructor parameters.

Reply:
Hmmm, this compiled for me on `2.13.5`:

```scala
  def bounded[F[+_]: Async: Dispatcher, A](requestedCapacity: Int)(implicit runtime: Runtime[Any]): F[Hub[F, A]] =
    ZHub.bounded[A](requestedCapacity).map(hub => CHub(hub)).toEffect
```

Reply:
Ahh perhaps - what didn't compile for me was the helper method called directly:
`toEffect(ZHub.bounded[A](requestedCapacity).map(hub => CHub(hub)))`

Reply:
This compiles for me as well:

```scala
  def bounded[F[+_]: Async: Dispatcher, A](requestedCapacity: Int)(implicit runtime: Runtime[Any]): F[Hub[F, A]] =
    toEffect(ZHub.bounded[A](requestedCapacity).map(hub => CHub(hub)))
```

Reply:
Ah they don't compile if we import `import zio.interop.catz._` instead of only `import zio.interop.catz.zManagedSyntax`

Reply:
2.12 also has a problem: https://github.com/zio/interop-cats/pull/263/checks?check_run_id=2388227935

Reply:
Oh I see. Because the `F` in `toEffect` being `ZIO[R, E, *]` is a valid solution as well.